### PR TITLE
Cm specials

### PR DESCRIPTION
**Pull request by chrismccord**
@feliperenan we'll need to decide ordering for the formatter, which I'm thinking will be `:let={} :for={} :if={}`

### CODE REVIEW

File: lib/phoenix_component.ex [hunk 1]
```diff
@@ -601,20 +601,53 @@ defmodule Phoenix.Component do
   `text_input`. The `Phoenix.Component` module has detailed documentation on
   how to use and implement such functionality.
 
-  #### :for
+  #### :if an d:for
```
Comment (josevalim) [inline lib/phoenix_component.ex:604]:
```suggestion
  #### :if and :for
```

File: lib/phoenix_live_view/html_engine.ex [hunk 2]
```diff
@@ -291,17 +293,30 @@ defmodule Phoenix.LiveView.HTMLEngine do
...
+            specials = Map.put(specials, key, special?)
+            {Map.put(acc_assigns, key, [assigns]), Map.put(acc_info, key, [info]), specials}
+        end
+      end)
+
+    # filter out nil slots via :if exclusion only when :if is present
+    acc_assigns =
+      Enum.into(acc_assigns, %{}, fn {key, assigns_ast} ->
+        if Map.fetch!(specials, key) do
+          {key, quote(do: Enum.flat_map(unquote(assigns_ast), & &1))}
```
Comment (josevalim) [inline lib/phoenix_live_view/html_engine.ex:317]:
```suggestion
          {key, quote(do: List.flatten(unquote(assigns_ast)))}
```

File: lib/phoenix_live_view/html_engine.ex [hunk 3]
```diff
@@ -956,54 +1017,53 @@ defmodule Phoenix.LiveView.HTMLEngine do
...
        )
-       when let in ["let", ":let"] do
-    quoted_value = Code.string_to_quoted!(value, line: line, column: col, file: file)
-    {{quoted_value, meta}, r, a, locs}
-  end
+       when attr in @special_attrs do
+    attr = if String.starts_with?(attr, ":"), do: attr, else: ":#{attr}"
+
+    case Map.fetch(special, attr) do
+      {:ok, {_, attr_meta}} ->
```
Comment (josevalim) [inline lib/phoenix_live_view/html_engine.ex:1051]:
```suggestion
    case special, do
      %{^attr => {_, attr_meta}} ->
```

File: lib/phoenix_live_view/html_engine.ex [hunk 4]
```diff
@@ -956,54 +1017,53 @@ defmodule Phoenix.LiveView.HTMLEngine do
...
+          column: attr_meta.column,
+          file: file,
+          description: message
 
-    raise ParseError,
-      line: meta.line,
-      column: meta.column,
-      file: file,
-      description: message
+      :error ->
```
Comment (josevalim) [inline lib/phoenix_live_view/html_engine.ex:1063]:
```suggestion
      %{} ->
```

**Review by josevalim — APPROVED** [review]

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**